task/backend/task_service_checkpointer.go

```diff
@@ -0,0 +1,47 @@
+package backend
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/influxdata/influxdb"
+)
+
+// Checkpointer allows us to restart a service from the last time we executed.
+type Checkpointer struct {
```

Comment:
This is a "task service checkpointer" meaning it implements the checkpointer interface but uses a task service to do it. A more specific name would be better here.